randovania/games/samus_returns/gui/preset_settings/msr_hints_tab.py

```diff
@@ -0,0 +1,43 @@
+import dataclasses
+
+from PySide6 import QtCore
+
+from randovania.game_description.game_description import GameDescription
+from randovania.games.samus_returns.layout.hint_configuration import ItemHintMode
+from randovania.gui.generated.preset_msr_hints_ui import Ui_PresetMSRHints
+from randovania.gui.lib.signal_handling import set_combo_with_value
+from randovania.gui.lib.window_manager import WindowManager
+from randovania.gui.preset_settings.preset_tab import PresetTab
+from randovania.interface_common.preset_editor import PresetEditor
+from randovania.layout.preset import Preset
+
+
+class PresetMSRHints(PresetTab, Ui_PresetMSRHints):
+    def __init__(self, editor: PresetEditor, game_description: GameDescription, window_manager: WindowManager):
+        super().__init__(editor, game_description, window_manager)
+        self.setupUi(self)
+
+        self.hint_layout.setAlignment(QtCore.Qt.AlignmentFlag.AlignTop)
+
+        for i, item_hint_mode in enumerate(ItemHintMode):
+            self.hint_artifact_combo.setItemData(i, item_hint_mode)
+
+        self.hint_artifact_combo.currentIndexChanged.connect(self._on_art_combo_changed)
+
+    @classmethod
+    def tab_title(cls) -> str:
+        return "Hints"
+
+    @classmethod
+    def uses_patches_tab(cls) -> bool:
+        return False
+
+    def _on_art_combo_changed(self, new_index: int):
+        with self._editor as editor:
+            editor.set_configuration_field(
+                "hints",
+                dataclasses.replace(editor.configuration.hints, artifacts=self.hint_artifact_combo.currentData()),
+            )
+
+    def on_preset_changed(self, preset: Preset):
+        set_combo_with_value(self.hint_artifact_combo, preset.configuration.hints.artifacts)
```

Comment:
Needs assertion for the type of `configuration` to access hints without mypy error.

Reply:
How important is this assertion? It looks like every other game that uses this exact function doesn't have this assertion. Might be from pre-mypy days though.

Reply:
Yeah, all other games are from pre-mypy days.
Mypy is weird at this because sometimes it's enough make an assertion on the field and sometimes it requires a new variable.

(besides the type import for `MSRConfiguration`)
```python
assert isinstance(preset.configuration, MSRConfiguration)
set_combo_with_value(self.hint_artifact_combo, preset.configuration.hints.artifacts)
```

or

```python
config = preset.configuration
assert isinstance(config, MSRConfiguration)
set_combo_with_value(self.hint_artifact_combo, config.hints.artifacts)
```

Reply:
Ah I was close but I had the wrong config part at the end